### PR TITLE
Add build phase to script - @jaredwy

### PR DESCRIPTION
**Pull request by califa**
This PR adds a build phase to the script.

---

There are a few things that are either wrong or could be improved: 

1. On lines 164 and 202, `TEMP_HTML_PATH` should be camelCased like the other constants in the file 
2. On line 168, the accidental assignment `=` should actually be a comparison `===`
3. Also on line 168, the opening conditional bracket `{` is missing 

Suggest the appropriate changes.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -155,6 +155,59 @@ async function main () {
...
+ * ==============================================================
+ *                         BUILD
+ * ==============================================================
+ */
+
+const TEMP_HTML_PATH = path.join(tempDir, inputFilenameNoExt + '_temp.htm')
+
+async function build (filepath) {
+  var shortFileName = filepath.replace(inputDir, '')
+  if ((path.basename(filepath) = 'config.yml') || (filepath.endsWith('.plugin.js')))
```
Comment (jaredwy) [inline script.js]: This line is missing a starting brace. 
```diff
-  if ((path.basename(filepath) = 'config.yml') || (filepath.endsWith('.plugin.js')))
+  if ((path.basename(filePath) = 'base.yml') || (filPath.endsWith('.test.js'))) {
```

File: script.js [hunk 2]
```diff
@@ -155,6 +155,59 @@ async function main () {
...
+ * ==============================================================
+ *                         BUILD
+ * ==============================================================
+ */
+
+const TEMP_HTML_PATH = path.join(tempDir, inputFilenameNoExt + '_temp.htm')
+
+async function build (filepath) {
+  var shortFileName = filepath.replace(inputDir, '')
+  if ((path.basename(filePath) = 'base.yml') || (filPath.endsWith('.test.js'))) {
```
Comment (jaredwy) [inline script.js]:
```diff
-  if ((path.basename(filePath) = 'base.yml') || (filPath.endsWith('.test.js'))) {
+  if ((path.basename(filePath) = 'base.yml') || (filePath.endsWith('.test.js'))) {
```

Comment (jaredwy) [inline script.js:168]:
```diff
-  if ((path.basename(filePath) = 'base.yml') || (filPath.endsWith('.test.js'))) {
+  if ((path.basename(filePath) = 'base.yml') || (filPath.endsWith('test.js'))) {
```

File: script.js [hunk 3]
```diff
@@ -155,6 +155,59 @@ async function main () {
...
+    console.log(colors.grey(`File ${shortFileName}: ignoring trigger, too busy.`))
+    return
+  }
+
+  console.log(colors.magenta.bold(`\nProcessing ${shortFileName}...`))
+  relaxedGlobals.busy = true
+  var t0 = performance.now()
+
+
+  var taskPromise = null
```
Comment (jaredwy) [inline script.js:192]:
```diff
-  var taskPromise = null
+  var taskPromise = {};
+  
```

**Review by jaredwy — CHANGES_REQUESTED** [review]

File: script.js [hunk 4]
```diff
@@ -155,6 +155,59 @@ async function main () {
...
+async function build (filepath) {
+  var shortFileName = filepath.replace(inputDir, '')
+  if ((path.basename(filePath) = 'base.yml') || (filPath.endsWith('.test.js'))) {
+    await updateConfig()
+    return
+  }
+  var page = relaxedGlobals.puppeteerPage
+  // Ignore the call if ReLaXed is already busy processing other files.
+
+  if (!(relaxedGlobals.watchedExtensions.some(ext => filepath.endsWith(ext)))) {
```
Comment (califa) [inline script.js:175]:
```diff
-  if (!(relaxedGlobals.watchedExtensions.some(ext => filepath.endsWith(ext)))) {
+  if (!(relaxedGlobals.wtensions.some(ext => filepath.endsWith(ext)))) {
```

**Pull request closed without merging**